modules/api/src/main/java/org/apache/ignite/table/QualifiedName.java

```diff
@@ -0,0 +1,306 @@
+
+        verifySchemaName(normalizedSchemaName);
+        verifyObjectName(normalizedObjectName);
+
+        return new QualifiedName(normalizedSchemaName, normalizedObjectName);
+    }
+
+    private static String parseIdentifier(String name) {
+        if (name == null || name.isEmpty()) {
+            return name;
+        }
+
+        var tokenizer = new Tokenizer(name);
+
+        String parsedName = tokenizer.nextToken();
+
+        if (tokenizer.hasNext()) {
+            throw new IllegalArgumentException("Non-qualified identifier expected: " + name);
+        }
+
+        return parsedName;
+    }
+
+    private static void verifyObjectName(@Nullable String name) {
+        if (name == null || name.isEmpty()) {
+            throw new IllegalArgumentException("Object name can't be null or empty.");
+        }
+    }
+
+    private static void verifySchemaName(@Nullable String schemaName) {
+        if (schemaName != null && schemaName.isEmpty()) {
+            throw new IllegalArgumentException("Schema name can't be empty.");
+        }
+    }
+
+    private final @Nullable String schemaName;
+    private final String objectName;
+
+    /**
+     * Constructs a qualified name.
+     *
+     * @param schemaName Normalized schema name.
+     * @param objectName Normalized object name.
+     */
+    private QualifiedName(@Nullable String schemaName, String objectName) {
+        this.schemaName = schemaName;
+        this.objectName = objectName;
+    }
+
+    /** Returns normalized schema name. */
+    public @Nullable String schemaName() {
+        return schemaName;
+    }
+
+    /** Returns normalized object name. */
+    public String name() {
+        return objectName;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+        if (object == null || getClass() != object.getClass()) {
+            return false;
+        }
+        QualifiedName that = (QualifiedName) object;
+        return Objects.equals(schemaName, that.schemaName) && Objects.equals(objectName, that.objectName);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        return Objects.hash(schemaName, objectName);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public String toString() {
+        String simpleName = IgniteNameUtils.quoteIfNeeded(objectName);
+
+        return schemaName == null ? simpleName : IgniteNameUtils.quoteIfNeeded(schemaName) + '.' + simpleName;
+    }
+
+    /**
+     * Identifier chain tokenizer.
+     *
+     * <p>Splits provided identifier chain (complex identifier like PUBLIC.MY_TABLE) into its component parts.
+     *
+     * <p>This tokenizer is not SQL compliant, but it is ok since it used to retrieve an object only. The sole purpose of this tokenizer
+     * is to split the chain into parts by a dot considering the quotation.
+     */
+    private static class Tokenizer {
```

Comment:
Why we can't use `IgniteNmeUtils.Tokenizer` ?

Reply:
It has wrong implementation.